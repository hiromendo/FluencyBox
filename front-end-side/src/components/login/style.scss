.base-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 15px;

  .header {
    font-size: 24px;
  }

  .content {
    display: flex;
    flex-direction: column;

    .image {
      width: 21em;
      img {
        height: 128px;
      }
    }

    .form {
      margin-top: 1em;
      display: flex;
      flex-direction: column;
      align-items: center;

      a {
        color: white;
      }

      .form-group {
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        width: fit-content;
        label {
          font-size: 20px;
          span {
            font-size: 0.85rem;
          }
        }
        input {
          font-family: inherit;
          margin-top: 6px;
          min-width: 18em;
          height: 37px;
          padding: 0px 10px;
          font-size: 16px;
          background-color: #f3f3f3;
          border: 0;
          border-radius: 4px;
          margin-bottom: 2px;
          transition: all 250ms ease-in-out;
          &:hover {
            background-color: #ffffff;
            box-shadow: 0px 0px 14px 0.3px #0e81ce96;
          }

          &:focus {
            outline: none;
            box-shadow: 0px 0px 12px 0.8px #3474dbb2;
          }
        }
        img.preview-upload {
          // height: 100px;
          width: 250px;
        }
      }
    }
  }

  .footer {
    margin-top: 1em;
  }
}
