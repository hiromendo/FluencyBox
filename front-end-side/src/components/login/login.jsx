import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import loginImg from '../../login.svg';
import { getLogin } from '../../actions';
import "./style.scss";

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userName : 'test@msn.com',
      password: '12345678'
    }
  }

  handleSubmit = event => {
    event.preventDefault();
    const { userName, password } = this.state;
    const userInfo = { userName, password };
    const { history } = this.props;

    this.props.getLogin(userInfo, history);
  }

  handleInputChange = event => {
    const targetName = event.target.name;
    this.setState({
      [targetName]: event.target.value
    })
  }
  render() {
    const { userName, password } = this.state;
    return (
      <div className="base-container" ref={this.props.containerRef}>
        <div className="header">Login</div> 
        <div className="content">
          <div className="image">
            <img src={loginImg} alt="login" />
            <form onSubmit={this.handleSubmit} className="form">
              <div className="form-group">
                <label htmlFor="userName">User Name</label>
                <input 
                  type="text" 
                  name="userName" 
                  value={userName} 
                  onChange={this.handleInputChange} 
                  required
                />
              </div>
              <div className="form-group">
                <label htmlFor="password">Password</label>
                <input 
                  type="password" 
                  name="password" 
                  value={password} 
                  onChange={this.handleInputChange} 
                  required
                />
              </div>
            <div className="footer">
              <button type="submit" className="btn">Login</button>
            </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = ({ loading }) => ({
  loading
})

const mapDispatchToProps = {
  getLogin
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login))