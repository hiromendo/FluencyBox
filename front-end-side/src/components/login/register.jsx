import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { sendRegister, cacheRegisterInfo, updateUserInfo } from '../../actions';
import "./style.scss";

class Register extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      userName: '',
      email: '',
      phone: '',
      password: '',
      confirmPassword: ''
    }
  }

  componentDidMount() {
    // this.generateRandomID()
    const { registerCache, reUpdateInfoProfile, authInfo: { serverResponse : { user } } } = this.props;
    if (reUpdateInfoProfile) {
      const obj = {
        firstname: user.first_name,
        lastname: user.last_name,
        userName: user.user_name,
        email: user.email_address,
        phone: user.phone_number,
      };
      this.setState(() => {
        return obj
      })

    } else {
      this.setState(prevState => {
        return registerCache
      })

    }
  }

  /* Dev purpose only */
  generateRandomID = () => {
    const c = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
    const randomID = Array.from({length:4}, _ => c[Math.floor(Math.random()*c.length)]).join('')
    this.setState({
      userName: randomID,
      email: `${randomID}@gmail.com`
    })
  }

  handleSubmit = event => {
    event.preventDefault();
    const { firstname, lastname, userName, password, phone, email, confirmPassword } = this.state;
    const { history, reUpdateInfoProfile, authInfo: { serverResponse: { user } } } = this.props;
    this.props.cacheRegisterInfo(this.state)
    if (reUpdateInfoProfile) {

      const userInfo = {
        "first_name" : firstname,
        "last_name" : lastname,
        "user_name" : userName,
        "email_address" : email,
        "phone_number" : phone,
      }

      this.props.updateUserInfo(userInfo, user.uid, history)
  
    } else {
      const isPasswordMatched = this.validatePassword()
      if (isPasswordMatched) {
        const userInfo = {
          "first_name" : firstname,
          "last_name" : lastname,
          "user_name" : userName,
          "email_address" : email,
          "phone_number" : phone,
          "password" : password,
          "confirm_password": confirmPassword
        }
        this.props.sendRegister(userInfo, history);

      }
    }
  }

  handleInputChange = event => {
    const targetName = event.target.name;
    const passWordInputNode = this.refs.passwordNode
    const confirmPasswordInputNode = this.refs.confirmPasswordNode;
    if (targetName === 'confirmPassword' && event.target.value.length >= 8 && this.refs.passwordNode.value) {
      if (event.target.value !== passWordInputNode.value) {
        confirmPasswordInputNode.setCustomValidity('Password does not match')
      } else {
        confirmPasswordInputNode.setCustomValidity('')
      }
    }
    this.setState({
      [targetName]: event.target.value
    })
  }

  validatePassword() {
    const { password, confirmPassword } = this.state;
    if (password !== confirmPassword) {
      return false
    } else {
      return true
    }
  }

  renderPassWordInputs() {
    const { reUpdateInfoProfile } = this.props;
    const { password, confirmPassword } = this.state;
    if (!reUpdateInfoProfile) {
      return (
        <>
          <div className="form-group">
            <label htmlFor="password">Password <span>8 minimum characters</span></label>
            <input 
              type="password" 
              name="password"
              ref="passwordNode"
              value={password}
              title="8 characters minimum"
              pattern=".{8,}"
              onChange={this.handleInputChange} 
              required
            />
          </div>
          <div className="form-group">
            <label htmlFor="confirmPassword">Confirm Password</label>
            <input 
              type="password" 
              name="confirmPassword"
              ref="confirmPasswordNode"
              value={confirmPassword}
              title="8 characters minimum"
              pattern=".{8,}"
              onChange={this.handleInputChange} 
              required
            />
          </div>
        </>
      )
    }
  }

  render() {
    const { firstname, lastname, userName, phone, email } = this.state;
    const { reUpdateInfoProfile } = this.props;

    return (
      <div className="base-container" ref={this.props.containerRef}>
        <div className="header">{reUpdateInfoProfile ? 'Update Profile Info' : 'Register'}</div> 
        <div className="content">
          <div className="image">
            <form onSubmit={this.handleSubmit} className="form">
              <div className="form-group">
                <label htmlFor="firstname">First Name</label>
                <input 
                  type="text" 
                  name="firstname" 
                  value={firstname}
                  onChange={this.handleInputChange} 
                  required
              />
              </div>
              <div className="form-group">
                <label htmlFor="lastname">Last Name</label>
                <input 
                  type="text" 
                  name="lastname" 
                  value={lastname}
                  onChange={this.handleInputChange} 
                  required
              />
              </div>
              <div className="form-group">
                <label htmlFor="userName">User Name</label>
                <input 
                  type="text" 
                  name="userName" 
                  value={userName}
                  onChange={this.handleInputChange} 
                  required
              />
              </div>
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input 
                  type="email" 
                  name="email" 
                  value={email}
                  onChange={this.handleInputChange} 
                  required
                />
              </div>
              <div className="form-group">
                <label htmlFor="phone">Phone Number</label>
                <input 
                  type="tel" 
                  name="phone"
                  value={phone}
                  onChange={this.handleInputChange} 
                  placeholder='Optional'
                />
              </div>
              {this.renderPassWordInputs()}
              <div className="footer">
                <button type="submit" className="btn">{reUpdateInfoProfile ? 'Update' : 'Register'}</button>
              </div>
            </form>
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = ({ loading, registerCache, authInfo }) => ({
  loading,
  registerCache,
  authInfo
})

const mapDispatchToProps = {
  sendRegister,
  cacheRegisterInfo,
  updateUserInfo
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Register))