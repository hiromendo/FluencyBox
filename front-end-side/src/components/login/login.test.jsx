import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'; 

import { Login } from './login';

Enzyme.configure({ adapter: new Adapter() })

describe('Login component', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})