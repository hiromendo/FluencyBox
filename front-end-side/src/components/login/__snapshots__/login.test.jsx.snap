// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should match snapshot 1`] = `
<div
  className="base-container"
>
  <div
    className="header"
  >
    Login
  </div>
  <div
    className="content"
  >
    <div
      className="image"
    >
      <img
        alt="login"
        src="login.svg"
      />
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <label
            htmlFor="userName"
          >
            User Name
          </label>
          <input
            name="userName"
            onChange={[Function]}
            required={true}
            type="text"
            value="kenzom123@msn.com"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value="12345678"
          />
        </div>
        <a
          href="/resetpassword"
          onClick={[Function]}
        >
          Forgot Password?
        </a>
        <div
          className="footer"
        >
          <button
            className="btn"
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
